core/textile/bucket/bucket_file.go

```diff
@@ -61,6 +61,27 @@ func (b *Bucket) UploadFile(ctx context.Context, path string, reader io.Reader)
 	return result, root, nil
 }
 
+func (b *Bucket) DownloadFile(ctx context.Context, path string, reader io.Reader) (result path.Resolved, root path.Path, err error) {
```

Comment:
DownloadFile to me means bringing a file on the remote to a local file (GetFile), but this is doing the same that `UploadFile` except it is additionally notifying the syncer. Shouldn't we just modify UploadFIle?